0.9.0

jonschlinkert opened the pull request:
There is still work to do, but this is going to be a lot easier to use than the current verb! I'll add some descriptive comments soon....

mightyiam commented:
@jonschlinkert this is a huge change. I hope you're keeping this up. I have high hopes.

jonschlinkert commented:
> I hope you're keeping this up. I have high hopes.

I am indeed! very much so. It's not always easy to see the connection between related projects, but I'm making commits to the underlying libs that verb uses. I can't wait to share this next release with the world, it's pretty awesome if I do say so myself lol :)

jonschlinkert committed:
use variable for related list, move toc

jonschlinkert committed:
get plugins loading from verb config

jonschlinkert committed:
make variables consistent

jonschlinkert committed:
adds `ask` to cli.map

jonschlinkert committed:
remove redundant data

jonschlinkert committed:
adds `get` helper

jonschlinkert committed:
adds `toc` middleware

jonschlinkert committed:
load data inside task, before rendering

jonschlinkert committed:
setup verb defaults

jonschlinkert committed:
get config/cli defaults setup

jonschlinkert committed:
update docs

jonschlinkert committed:
update deps

jonschlinkert committed:
update readme

jonschlinkert committed:
add layouts, need to customize them still

jonschlinkert committed:
adds `ask` flag

jonschlinkert committed:
only add a layout if not already defined

jonschlinkert committed:
generate docs

jonschlinkert committed:
docs

jonschlinkert committed:
rename `runner.js` to preload

doowb committed:
add base-list dependency

doowb committed:
setup mappings to enable running base-list methods

doowb committed:
ensure entire `verbApps` configuration is run after any settings have been set
add displaying and/or choosing apps and tasks when flags are set.

jonschlinkert committed:
add related link to data.md docs

jonschlinkert committed:
plugins/middleware docs

jonschlinkert committed:
handle arrays of plugins

jonschlinkert committed:
improve error messages in verbfile

jonschlinkert commented:
@mightyiam you might be able to download this now and test is if you want :) 

**NOTE** that:

- `license` no longer a helper by default, it's just a property. You can make it a helper if you want, or just change `{%= license() %}` to `{%= license %}` in `.verb.md`
- I haven't gotten to the `apidocs` helper yet. I'll be doing that tomorrow. 

Also, if you (or anyone) wants to try this out and give feedback on an or all of the following that would be great:

- download verb: `npm i -g verb`
- run `verb` in the command line in a project that has a `.verb.md` (make sure you note the changes above)
- verb should prompt you for basic info 
- if you run `verb` again in that project, it should not prompt you again
- try doing `verb --init` in the same project
- try doing `verb --ask`, `verb --ask=name`, `verb --ask=name,description`
- then run `verb` again

Thanks!

jonschlinkert committed:
min layout/verb.md templates

jonschlinkert committed:
don't apply layouts to partials in `preLayout` middleware

jonschlinkert committed:
clean up

jonschlinkert committed:
generate readme

jonschlinkert committed:
update deps

mightyiam commented:
@jonschlinkert thank you

mightyiam commented:
Would you mind if I test after you release?

jonschlinkert commented:
No, of course not! I was just hoping for feedback from someone before I release. Thanks!

jonschlinkert committed:
start adding apidocs helper

jonschlinkert committed:
emit plugins, enqueue default readme questions

jonschlinkert committed:
organize cli mappings

jonschlinkert committed:
add helpers

jonschlinkert committed:
adds `updater` to udpate package.json properties

tunnckoCore commented:
My god.. (and i'm not religious guy lol)
You know you're awesome and verb is more awesome lol. Almost 2 years work on this things and so much libs. 
I'll try ASAP as I have stable config. I hope I'll be back soon, lol. Maybe this or next weekend.

jonschlinkert commented:
Thanks! yeah it is, but it all somehow comes together (almost) perfectly :) 

We found some bugs this morning, got most of them fixed, so anyone interested in playing around with this should wait until the next push. 

> verb is more awesome lol.

We think it's pretty great too! I look forward to feedback!

jonschlinkert committed:
don't use `get`, un-abstract

jonschlinkert committed:
get data, middleware and plugins loading as expected on all instances

jonschlinkert committed:
adds `expand` and `schema`, both wip

- schema will be used for validating verb config values
- expand will be used to convert string values to their object equivalents in memory (e.g. the `author` string in package.json will be expanded to an object for templates)

jonschlinkert committed:
fix delims

jonschlinkert committed:
add contributing template, needs to be updated

jonschlinkert committed:
adds `description` to related list sections

jonschlinkert committed:
rename `verbApps` to `apps`

jonschlinkert committed:
rename helper fn to `formatLicense`

jonschlinkert committed:
adds `expandConfig` util, to resolve templates in verb config

jonschlinkert committed:
a number of middleware improvements, adds `base-diff`

jonschlinkert committed:
move preload to `initVerb` method

jonschlinkert committed:
fix name from template in contributing.md

jonschlinkert committed:
lint

jonschlinkert committed:
clean up cli, scaffold out `init` and `help` mappings

jonschlinkert committed:
compartmentalize toc logic

jonschlinkert committed:
rename `userDocs` var to `userDir`

jonschlinkert committed:
start cleaning up verbfile

jonschlinkert committed:
updte deps

jonschlinkert committed:
lint

jonschlinkert committed:
generate readme

jonschlinkert committed:
dont test instances on cloned `views`

(since we already cover this in ancestor libs, and we will probably be getting a different ctor here)

jonschlinkert committed:
improve mappings for config/cli properties

jonschlinkert committed:
set config values from the command line

doowb committed:
adding `base-diff` to dependencies

jonschlinkert committed:
split out some commands

jonschlinkert committed:
move templates into `lib/templates`

jonschlinkert committed:
remove `full` layout since it was almost the same as `default`

start adding metadata to default templates

jonschlinkert committed:
move `ask` task to `cli.js`

jonschlinkert committed:
ensure TOC is generated and inserted only when specified by the user

jonschlinkert committed:
adds `require`, `apidocs` and `issue` helpers

jonschlinkert committed:
move contributing to templates

jonschlinkert committed:
only add twitter question if it doesn't exist

jonschlinkert committed:
remove list helper

jonschlinkert committed:
improve logic to determine if a table of contents is specified

jonschlinkert committed:
adds config mappings for helpers, middleware. several fixes

jonschlinkert committed:
expand package.json data before merging into context

jonschlinkert committed:
ensure that values are cast to the desired type

jonschlinkert committed:
remove `data` from `defaults.js`

jonschlinkert committed:
create `sections` partials, rename `libFiles` to `jsFiles`

jonschlinkert committed:
adds several utils for merging/flattening data

jonschlinkert committed:
cli improvements

jonschlinkert committed:
adds `pkg` plugin, for getting/setting values on package.json

jonschlinkert committed:
adds support for `sections`

jonschlinkert committed:
lint

jonschlinkert committed:
start working on sections, modularize code to start moving it out

jonschlinkert committed:
get `pkg` working

jonschlinkert committed:
another attempt to get TOC middleware working correctly

also adds examples middleware, to add examples to the comments context

jonschlinkert committed:
adds `actions` object

jonschlinkert committed:
adds `resolve` and `exists` helpers

jonschlinkert committed:
start adding section logic

jonschlinkert committed:
add base and env to args for `templates` util

jonschlinkert committed:
clean up questions junk

jonschlinkert committed:
update/move templates

jonschlinkert committed:
move some middleware functions into files, clean up

jonschlinkert committed:
externalize loader function

jonschlinkert committed:
don't use loader for built-in templates

jonschlinkert committed:
lint

jonschlinkert committed:
CLI utils

jonschlinkert committed:
namespace `sections` collection

jonschlinkert committed:
adds `read` and `example` helpers

jonschlinkert committed:
re-add diff middleware

jonschlinkert committed:
clean up data-expansion logic

jonschlinkert committed:
lint

jonschlinkert committed:
adds verb `sections`

jonschlinkert committed:
docs

jonschlinkert committed:
update readme template to use `sections`

jonschlinkert committed:
remove unused deps

jonschlinkert committed:
clean up docs

jonschlinkert committed:
CLI middleware updates

jonschlinkert committed:
fix error messages in applyLayout tests

jonschlinkert committed:
clean up toc middleware

jonschlinkert committed:
generate docs

jonschlinkert committed:
lint

jonschlinkert committed:
refactor

jonschlinkert committed:
update unit tests

jonschlinkert committed:
just silly. this needs to be refactored

jonschlinkert committed:
setup verb defaults

jonschlinkert committed:
lint and update deps

jonschlinkert committed:
clean up generators

jonschlinkert committed:
simplify cli

jonschlinkert committed:
move cli/config logic into base-runner and base-cli

jonschlinkert committed:
verbfile and readme

jonschlinkert committed:
update verbfile to use readme generator

jonschlinkert committed:
minor formatting

jonschlinkert committed:
inherit generate

jonschlinkert committed:
remove unused deps

jonschlinkert committed:
update tests

jonschlinkert committed:
fix helper and generate tests

jonschlinkert committed:
comments and error handling in `bin/verb.js`

jonschlinkert committed:
only create default collections when CLI is used

jonschlinkert committed:
remove unused deps, enable TOC in readme

jonschlinkert committed:
update docs

jonschlinkert committed:
use static `.runner` method from `generate`

jonschlinkert committed:
start adding CLI flags

jonschlinkert committed:
clean up

jonschlinkert committed:
custom `toFullname` function

jonschlinkert committed:
process.exit on `done`

jonschlinkert committed:
clean up verbfile

jonschlinkert committed:
generate docs

jonschlinkert committed:
reflinks

jonschlinkert committed:
update docs

jonschlinkert committed:
start adding cli docs

jonschlinkert commented:
@tunnckoCore if you have time and want to try this out, you can do:

```sh
$ npm i -g verbose/verb#dev && npm i -g verb-readme-generator
```

then run `verb readme` 

I think we're getting close, mostly bugfixes at this point. but the generators are awesome.

jonschlinkert commented:
if you do, let me know how that goes. same with `generate`, if you want to play around with it, you can run generators with the `gen` command. so `gen foo` would run the foo generator. to run a sub-generator on `foo` you would do `gen foo.bar`

tunnckoCore commented:
@jonschlinkert thanks for the ping!
I'll try later today after the exam.
But by quick look it seems `verb-readme-generator` currently not exist in npm. :)

jonschlinkert commented:
> But by quick look it seems verb-readme-generator currently not exist in npm. :)

it does now ... lol

jonschlinkert commented:
btw, [this file is a generator](https://github.com/verbose/verb-readme-generator/blob/master/index.js), in case you want to see a working example. it's has a lot in it, some things might get moved out, but it's a good example

tunnckoCore commented:
Yea, noticed. Btw, i think you should set default name of readme to be `README.md` it is more visible and widely used. We want sane defaults, yea? :)

But yea other than that, great! atm

tunnckoCore commented:
@jonschlinkert strange. Gives me ENOENT error that `bin/verb.js` missing, but it's there when look at repo, and it also is correct in package.json. Any ideas? hmm.

doowb commented:
@tunnckoCore did you remove any previously globally installed `verb` and `verb-cli` modules?

Try doing:

```sh
$ npm uninstall -g verb
$ npm uninstall -g verb-cli
$ npm i -g verbose/verb#dev
```

If you get permission errors on any of those, try going into the global node `bin` folder and delete the symlinks manually. I don't know if this happened to me because I'm using `nvm`, but everything installed correctly ensuring all the previous versions were deleted before trying to install the newest.

tunnckoCore commented:
> did you remove any previously globally installed verb and verb-cli modules?

Yea, did it.

> If you get permission errors on any of those, try going into the global node bin folder and delete the symlinks manually.

But yea, it's interesting that npm not exits with perm errors, but with enoent. And yea, there's no symlinks.

doowb committed:
including bin and lib in files

doowb commented:
@tunnckoCore try installing now with `npm i -g verbose/verb#dev`. The `bin` and `lib` folders were not in the `package.json` files prop. I think I didn't have the issue because I was doing `npm link` locally.

jonschlinkert commented:
like @doowb said, I sometimes have to go into both global `node_modules` and global `bin` and delete symlinked modules. if you did that and it didn't work, I'm not sure what's causing that.

jonschlinkert commented:
> I think I didn't have the issue because I was doing npm link locally.

it's @doowb's fault then lol. jk

tunnckoCore commented:
oo yea, that's the problem, i missed to look `files` field.

tunnckoCore commented:
Okey it works now.

> @jonschlinkert: then run verb readme

error, lol.

```js
[charlike@voltaire trans]$ verb readme
[20:37:22] starting default generator
GeneratorError {
  message: 'task "readme" is not registered',
  name: 'GeneratorError: generator [default] reason',
  stack: 'GeneratorError: generator [default] reason: task "readme" is not registered\n    at new GeneratorError (/usr/local/lib/node_modules/verb/node_modules/base-generators/lib/cache.js:190:17)\n    at Cache.error (/usr/local/lib/node_modules/verb/node_modules/base-generators/lib/cache.js:171:16)\n    at Cache.<anonymous> (/usr/local/lib/node_modules/verb/node_modules/base-generators/lib/cache.js:117:40)\n    at Verb.Emitter.emit (/usr/local/lib/node_modules/verb/node_modules/component-emitter/index.js:131:20)\n    at finishBuild (/usr/local/lib/node_modules/verb/node_modules/composer/index.js:143:12)\n    at /usr/local/lib/node_modules/verb/node_modules/composer/lib/flow.js:22:16\n    at Verb.Composer.build (/usr/local/lib/node_modules/verb/node_modules/composer/index.js:151:10)\n    at Verb.<anonymous> (/usr/local/lib/node_modules/verb/node_modules/base-generators/index.js:454:13)\n    at /usr/local/lib/node_modules/verb/node_modules/async/lib/async.js:52:16\n    at /usr/local/lib/node_modules/verb/node_modules/async/lib/async.js:269:32' }
{}

```

tunnckoCore commented:
Maybe it tries to load given generator from local modules because i tried with verbfile.js from the readme

```
[charlike@voltaire trans]$ verb
[20:41:49] using verbfile ~/dev/verbfile.js
[Error: cannot find generator: verb-readme-generator]
```

jonschlinkert commented:
I've reinstalled a couple of times and cleared node_modules locally and globally and I can't reproduce the issue you're having.

tunnckoCore commented:
Strange.

```
[charlike@voltaire dev]$ ls /usr/local/lib/node_modules/
npm
[charlike@voltaire verbtest]$ npm i -g verbose/verb#dev verb-readme-generator
[charlike@voltaire verbtest]$ ls /usr/local/lib/node_modules/
npm  verb  verb-readme-generator
[charlike@voltaire verbtest]$ verb readme
[18:15:44] starting default generator
GeneratorError {
  message: 'task "readme" is not registered',
  name: 'GeneratorError: generator [default] reason',
  stack: 'GeneratorError: generator [default] reason: task "readme" is not registered\n    at getTaskFn (/usr/local/lib/node_modules/verb/node_modules/composer/lib/resolve.js:50:11)\n    at Verb.module.exports (/usr/local/lib/node_modules/verb/node_modules/composer/lib/resolve.js:35:13)\n    at /usr/local/lib/node_modules/verb/node_modules/composer/lib/flow.js:20:23\n    at Verb.Composer.build (/usr/local/lib/node_modules/verb/node_modules/composer/index.js:151:10)\n    at Verb.<anonymous> (/usr/local/lib/node_modules/verb/node_modules/base-generators/index.js:456:13)\n    at /usr/local/lib/node_modules/verb/node_modules/async/lib/async.js:52:16\n    at /usr/local/lib/node_modules/verb/node_modules/async/lib/async.js:269:32\n    at /usr/local/lib/node_modules/verb/node_modules/async/lib/async.js:44:16\n    at MapConfig.<anonymous> (/usr/local/lib/node_modules/verb/node_modules/map-config/index.js:163:14)\n    at Immediate.iterate [as _onImmediate] (/usr/local/lib/node_modules/verb/node_modules/async/lib/async.js:262:13)' }
{}
```
Okey, what should happen with `verb readme` exactly? Read `.verb.md` (i'm understand that it's not needed when using `verb-readme-generator`, i think that's the point of it, or?) , ask, populates and generates `readme.md` or? Can you screen with https://asciinema.org?

jonschlinkert commented:
> (i understand that it's not needed when using verb-readme-generator, i think that's the point of it, or?)

no, that might be the problem. you don't need a `verbfile.js`, but you do need a `.verb.md`

tunnckoCore commented:
> no, that might be the problem.

Nope. Because it errors thar there's no readme task, so it can't find `verb-readme-generator`. No matter if I have `.verb.md` or not, it fails again.

jonschlinkert commented:
ok, let's get this solved. please double check these and make sure they're all correct. it's totally fine if you missed something... it happens to us constantly when we publish something new. 

None of these are "correct" or "wrong", and you gave info on some of them already, but I want to double check so we can learn how you're actually using it, not how you think I "want you to". That will help me make this experience better for everyone:

- [ ] did you delete all `verb` related files from the **global** `bin` directory? if not, please do this. It's not sufficient to run npm update, or `cache clean`. again, make sure `verb-cli` is deleted too
- [ ] did you delete all `verb` related files from the **global** `node_modules` directory? these first two are the most likely causes, even though you said you cleared them. many times I've thought the same thing and realized I hadn't afterwards.
- [ ] is verb installed globally or locally?
- [ ] is `verb-readme-generator` installed in `devDependencies` of the project you created? or is it installed globally? or not at all?
- [ ] does the project have a package.json? (if not, we'll need to make sure verb doesn't have a problem with that)
- [ ] if so, does the package.json have a `verb` config object in it? and if so, can you paste it here?
- [ ] do you by any chance have a global `verb` config store in your user home? if so, it's possible that it's using old settings that are conflicting (no idea, just grasping for answers)
- [ ] are you using a `verbfile.js`? you don't need one to run verb or the `readme` task

**if yes***

- [ ] does it export a function or an instance of verb? either _should_ work
- [ ] is `verb-readme-generator` invoked (with `.extendWith()` at the top of the exported function?

**if no**

- [ ] are you changing the `cwd`, or passing any other flags besides the `verb readme` command?

can you paste your verbfile and package.json in a gist? or provide any other information that might help us debug? 

honestly we were both laughing (not at you, at the situation) b/c this seems to happen every time. it seems like maybe there's something that we're doing that makes our setup "just work", or something you're doing that makes it "not work".  

seriously, given how consistent it is that this happens, I think it's a real possibility. I'd love to figure out what the problem is so we don't have to keep going back and forth, and so we can spare other users from the same agony lol.

tunnckoCore commented:
Yes I deleted all from `bin` and `node_modules` directories. Cache cleaned. And actually uninstalled all of the global modules, as you can see in above output - there was only `npm`. Then, i just installed `verb` and `verb-readme-generator` **only** globally as [you suggest](https://github.com/verbose/verb/pull/78#issuecomment-183216052) firstly.

Recored screen on asciinema https://asciinema.org/a/dzk7o7sss0fj7xs3t0fnblupk

I don't have any global or local verb configs. Also there's no `.verb.md` and can't get the point why it could be needed when using `verb-readme-generator` - as i seen yesterday it have some templates and default layouts?

jonschlinkert commented:
> with locally installed verb-readme-generator and custom .verb.md it seems it works.

lol I said that earlier though! you need a `.verb.md` locally.

> what's the point

I just haven't gotten to that yet lol. one step at a time!

tunnckoCore commented:
> I just haven't gotten to that yet lol.

i got it.

I'm now playing. But don't like the idea that should always install `verb-readme-generator` locally in every project which is currently working flow, it not work if it is only globally.

jonschlinkert commented:
> But don't like the idea that should always install verb-readme-generator locally in every project which is currently working flow,

you don't need to. you never need to use it unless you want to create a generate that extends it. I just wanted to make sure that was working. you can just install it globally and do `verb readme`, or do `verb ---config=tasks:readme` and it will automatically run that generator on that project, or do `verb --save=tasks:readme` and it will run the readme generator for every project.

tunnckoCore commented:
> you can just install it globally

that's what i'm talking about (so i understand it correctly from the beginning) it not fails it above error if it is installed only globally.

jonschlinkert committed:
update tests

jonschlinkert committed:
remove variables in docs

jonschlinkert committed:
lint tests

jonschlinkert committed:
run update

jonschlinkert committed:
tweak aliasFn and fullnameFn

jonschlinkert committed:
update deps

jonschlinkert committed:
remove junk from lib

jonschlinkert committed:
reorganize init code to ensure objects exist when they need to

jonschlinkert committed:
questions tests

jonschlinkert committed:
clean up lib

jonschlinkert committed:
adds debug

jonschlinkert committed:
update deps

jonschlinkert committed:
generate docs

jonschlinkert commented:
@tunnckoCore were you able to get verb installing globally?

tunnckoCore commented:
Yea, now it works. :) Also globally installed generators works :) Great!

jonschlinkert committed:
update tests

jonschlinkert committed:
adds `isGeneratorPath`

jonschlinkert committed:
generate docs

jonschlinkert committed:
start scaffolding out some docs for middleware

jonschlinkert committed:
test fixtures

jonschlinkert committed:
start on config docs

jonschlinkert committed:
generate docs

jonschlinkert committed:
update deps

jonschlinkert committed:
update tests to reflect dependency changes

jonschlinkert committed:
change default generator to display the `--help` menu

jonschlinkert committed:
generate docs

jonschlinkert committed:
update deps

jonschlinkert committed:
don't create docs collection

jonschlinkert committed:
generate docs

jonschlinkert committed:
update deps

ericclemmons commented:
Crap. Merge conflicts.  A good ol' `git push origin master --force` should do the trick! ;)

jonschlinkert commented:
@tunnckoCore, @doowb what do you think? is verb 0.9.0 ready to merge to master? It works consistently for me, how has your experience been? what major issues do you think need to be addressed?

doowb commented:
>  is verb 0.9.0 ready to merge to master?

I think yes!

> how has your experience been?

So far it's been great in projects that have an old `.verb.md` file, I'm able to run `verb --init` to add the common configuration settings to the `package.json` then run `verb`.

> what major issues do you think need to be addressed?

I haven't run into major issues recently. There might be something minor that I've overlooked due to the _curse of knowledge_ but those should surface with new eyes.

jonschlinkert commented:
nice, thanks @doowb!

tunnckoCore commented:
Oh yes! It's looks and feels great! I think it's ready. I'll just check what commit i'm using, I think it is the latest, but need recheck.

jonschlinkert commented:
great! I have a few thing to push up tomorrow, and if possible I'll merge but it might be this weekend. then docs are the number 1 priority.

tunnckoCore commented:
> then docs are the number 1 priority.

I'll try to help there. :)

jonschlinkert committed:
draft docs

jonschlinkert committed:
fix label

jonschlinkert committed:
format table

jonschlinkert committed:
clarify config types

jonschlinkert committed:
organize docs

jonschlinkert committed:
minor formatting

jonschlinkert committed:
add ctor events, improve alias regex

jonschlinkert committed:
minor edits

jonschlinkert committed:
remove listener

jonschlinkert committed:
run update

jonschlinkert committed:
adds custom commands

jonschlinkert committed:
simply

jonschlinkert committed:
use base-runner

jonschlinkert committed:
add default tasks

jonschlinkert committed:
update tests

jonschlinkert committed:
update patterns, lint gulpfile

jonschlinkert committed:
add `questions` for init

jonschlinkert committed:
update deps, scaffold out `reflinks`, `format` (WIP)

jonschlinkert committed:
draft recipe for extending a generator

jonschlinkert committed:
generate readme

jonschlinkert committed:
remove wip files

jonschlinkert committed:
adds `toAlias` function

jonschlinkert committed:
generate docs

jonschlinkert committed:
update deps

jonschlinkert committed:
adds support for `--format`

jonschlinkert committed:
reflect changes from base-generators

jonschlinkert committed:
remove unused code from lib

jonschlinkert committed:
reflect changes from base-runner

jonschlinkert committed:
lint

jonschlinkert committed:
update verb config, description

jonschlinkert committed:
add/update vinyl-fs tests

jonschlinkert committed:
use `verb-readme-generate` full name as default

so that in the case there are errors it's less confusing

jonschlinkert committed:
update error messages in tests

jonschlinkert committed:
add sub-generators and tasks to `defaults`

jonschlinkert committed:
fix pattern used for lookups

jonschlinkert committed:
adds `gulp-unused`

jonschlinkert committed:
add verbfile.js for docs

jonschlinkert commented:
@tunnckoCore hey thanks for the mentions on other issues. we'll need some help getting the word out once this version is released.

jonschlinkert committed:
update unit tests

jonschlinkert committed:
temporarily remove docs

jonschlinkert committed:
update default templates to ensure no layout is used

jonschlinkert committed:
use `.generate`

jonschlinkert committed:
move `contributing.md` to `.github` folder

jonschlinkert committed:
update lookup patterns, init events

jonschlinkert committed:
suggest default layout based on lib

jonschlinkert committed:
adds render task, diffs

jonschlinkert committed:
adds macro support, better task handling

jonschlinkert committed:
adds default templates

jonschlinkert committed:
remove unused var

jonschlinkert committed:
update deps

jonschlinkert committed:
logger, ctor events

jonschlinkert committed:
update tests

jonschlinkert committed:
run update

doowb committed:
deps

doowb committed:
typo

jonschlinkert committed:
adds version command

jonschlinkert committed:
change lookup pattern to `verb-generate-*`

jonschlinkert committed:
adds logo

jonschlinkert committed:
update verbfile

jonschlinkert committed:
update deps

jonschlinkert committed:
update unit tests

jonschlinkert committed:
docs

jonschlinkert committed:
run update

jonschlinkert committed:
add format plugin

jonschlinkert committed:
improve template

jonschlinkert committed:
add list task to built-in generator

jonschlinkert committed:
make defaults smarter

jonschlinkert committed:
use yargs-parser directly

jonschlinkert committed:
update deps

jonschlinkert committed:
generate docs

jonschlinkert committed:
add verb stores

jonschlinkert committed:
clean up reflinks in package.json

jonschlinkert committed:
lint

jonschlinkert committed:
start scaffolding out `init` task

jonschlinkert committed:
progress on docs

jonschlinkert committed:
previewing docs

jonschlinkert committed:
fix branch

ierceg committed:
Fix bad syntax in verbfile.js template

jonschlinkert committed:
Merge pull request #105 from ierceg/ierceg-patch-1

Fix bad syntax in verbfile.js template

adoyle-h commented:
How is the project going on? I have waited a long time for the new release.

tunnckoCore commented:
@adoyle-h just use that branch. No need to wait, this PR is about years old and couple of users already uses 0.9 from the beginning.

jonschlinkert commented:
prs and docs are welcome!

aredshaw committed:
mostly fixed links and deleted those without a reference I could find

doowb committed:
Merge pull request #113 from aredshaw/updated-dev-verb

mostly fixed links and deleted those without a reference I could find

jonschlinkert committed:
update dotfiles

jonschlinkert committed:
lint

jonschlinkert committed:
update verb
